### ISSUE TITLE
Move default building options back to the preferences menu

### ISSUE THREAD

**Issue by SkuliAdams** [ClaudeMetz/FactoryPlanner#366]
### Suggestion

Updating to 2.0, I was somewhat confused about several options disappearing from the preferences menu. I ultimately found where to set a building to the default building, but the new location is unintuitive. For one, changing defaults now requires more steps, needing to find a recipe of the specific category you want to change the default for before setting the default. Additionally, it is no longer possible to manually set defaults across many categories. Before, I could just go down the list and click on my preferred building for each one, but now you have to do it one at a time. This is especially confusing with buildings present in multiple categories; for example, if you set the assembly machine 2 to the default using the inserter recipe, creating a new circuit recipe in the planner defaults to the assembly machine 1 because it's a different category.

In short, having one place to set all building defaults like prior to 2.0 is significantly easier to use, so this should be re-added, possibly in addition to the current method location rather than a complete revert to the old method.

**Comment by RNO-HMRKG** [comment]
Well, thanks to @SkuliAdams i now know where to set it in the first place.
For everyone looking: Select Receip -> Click on the Assembling Machine -> Set the new Machine -> Same window there is now a "Defaults" where you can choose to 1: "Set as Default" 2: "Set for all Categories"

@SkuliAdams isn't the "Set for all Categories" quite the functionality you are requesting? I found it quite cumbersome to do it the old way if I think about it now. I like the new way even if its a bit hidden.
![grafik](https://github.com/user-attachments/assets/0ed76b7e-7451-4a47-94ac-b47b637f3157)

**Comment by SkuliAdams** [comment]
While that does help somewhat, it still has the problem of not telling the player *what* the relevant categories are, and comes with the risk of the player setting a default they didn't intend to. Hence why I suggest using the old method as well, so an overview of all categories is available.

**Comment by ClaudeMetz** [comment]
I'll probably have both options at some point.

**Comment by ClaudeMetz** [comment]
Yeah I get that, it's just that the settings have become more complicated now, with machines and fuels with multiple categories, etc. So the UI would be much more cluttered nowadays. I still want to look into this and see if I can come up with a good UI for it though.

**Comment by ClaudeMetz** [comment]
This comes up now and again, and it makes sense why people think they want it. As I explain the complications though, they all tend to realize why it wouldn't really work. So I don't think this is coming back.

I'm leaving the issue open though, as I could probably improve the machine/beacon dialog UI to help with discoverability.

**Comment by ClaudeMetz** [comment]
This is just not happening. It's messier than people asking for it think, and it wouldn't help them as much as they expect. What could happen though is #784, allowing you to propagate your choice elsewhere. Not really addressing the same problem, but eh. I will also try to look into making the machine/beacon UI better potentially, as it's often missed now. Not sure what that'll look like though.
